src/app/(auth)/subscriber/edit/page.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+
+import SubscriberForm from '@/components/Subscriber/SubscriberForm';
+import { UserGroupIcon } from '@heroicons/react/24/outline';
+import { updateSubscriber, getSubscriber } from '@/apis/subscriber';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { useRouter, useSearchParams } from 'next/navigation';
+import {
+  mapTagsToSelectOptions,
+  extractValuesFromOptions,
+} from '@/utils/common';
+
+const Page = () => {
+  const searchParams = useSearchParams();
+  const id = searchParams.get('id');
+  const router = useRouter();
+  const [selectedOptions, setSelectedOptions] = useState([]);
+  const [subscriber, setSubscriber] = useState({});
+
+  useEffect(() => {
+    const fetchSubscriber = async () => {
+      try {
+        const { data } = await getSubscriber(id);
+        setSelectedOptions(mapTagsToSelectOptions(data.tags));
```

Comment:
since we're passing the whole `subscriber` object to the form, the values that need to be selected in a field in the form should be a responsibilty of the form and not the parent component.